### DISCUSSION TITLE
How can the WGPU trace be parsed?

### DISCUSSION THREAD

**Discussion by davids91** [bevyengine/bevy#13960] [Q&A] [answered]
I've followed the guide here: https://github.com/bevyengine/bevy/blob/main/docs/profiling.md
I added the relevant trace features to the cargo file: 
`bevy = { version = "0.13.2", features = ["dynamic_linking","trace", "trace_chrome", "wgpu_trace"], optional = true}`
so now I get the json file generated, and also there is a wgpu_trace folder. 

How can the contents be displayed? 
[ui.perfetto.dev ](https://ui.perfetto.dev/) does not open it, and I couldn't find any info on this while googling..

**Comment by JMS55** [comment, marked as answer]
wgpu_trace is mostly for debugging purposes. It's not meant for profiling your app. Bevy's profiling guide does not mention it.

Tracy will give you CPU timing spans. A sampling profiler (e.g. perf) will give you more detailed CPU performance info. GPU timestamps will give you GPU timing spans. GPU profilers (e.g. NSight or RGP) will give you more detailed GPU performance info.

**Reply by davids91** [reply]
That's not what I was hoping for, but at least I suspected it... '^^ Thank you for the answer!
